crates/runtime/src/instance.rs

```diff
@@ -1246,12 +1278,15 @@ fn initialize_tables(instance: &Instance) -> Result<(), InstantiationError> {
         }
 
         for (i, func_idx) in init.elements.iter().enumerate() {
-            let anyfunc = instance.get_caller_checked_anyfunc(*func_idx);
+            let anyfunc = instance.get_caller_checked_anyfunc(*func_idx).map_or(
+                instance.null_func_ref(),
+                |f: &VMCallerCheckedAnyfunc| {
+                    let p = f as *const VMCallerCheckedAnyfunc as *mut _;
+                    unsafe { NonNull::new_unchecked(p) }
```

Comment:
I think there's a safe conversion from `&T` to `NonNull<T>` to use here?

Reply:
I didn't realize that `NonNull<T>` implements `From<&T>`!